tests/ssl-opt.sh

```diff
@@ -2954,6 +2981,207 @@ run_test    "Authentication: send CA list in CertificateRequest, client self sig
             -c "! mbedtls_ssl_handshake returned" \
             -s "X509 - Certificate verification failed"
 
+# Tests for auth_mode, using CA callback
```

Comment:
I think this section is a near-duplicate of the previous one, which is fine but should probably be documented here and in the previous section, so that in the future when we update one we remember to update the other accordingly.

Reply:
I think that the need to duplicate the tests shows that the test system should be reworked to avoid that.